### PR TITLE
Stress Testing

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/rancher/issues/9110

### CODE REVIEW

**Comment by moelsayed** [comment]
@sangeethah This is ready for review.

File: tests/v2_validation/cattlevalidationtest/core/common_fixtures.py [hunk 1]
```diff
@@ -2957,7 +2986,8 @@ def add_digital_ocean_hosts(client, count, size="2gb",
     if docker_version == "1.10":
         os_version = "ubuntu-14-04-x64"
     for i in range(0, count):
-        create_args = {"hostname": random_str(),
+        # need extra random names for this one(sorry).
+        create_args = {"hostname": random_str() + str(random_num()),
```
Comment (sangeethah) [inline tests/v2_validation/cattlevalidationtest/core/common_fixtures.py]: I am not too sure if making this extra random is going to solve the issue you are seeing with DO machine failures.

File: tests/v2_validation/cattlevalidationtest/core/test_stress.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+    check_round_robin_access_lb_ip(pod_new_names, lbips[0], port,
+                                   hostheader="foo.bar.com",
+                                   path="/service3.html")
+
+    check_round_robin_access_lb_ip(["nginx-ingress2"], lbips[0], port,
+                                   hostheader="foo.bar.com",
+                                   path="/name.html")
+
+
```
Comment (sangeethah) [inline tests/v2_validation/cattlevalidationtest/core/test_stress.py]: We should be able to reuse the validate_stack() from test_upgrade.py

File: tests/v2_validation/cattlevalidationtest/core/test_stress.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    return result
+
+
+def render(tpl_path, context):
+    path, filename = os.path.split(tpl_path)
+    return jinja2.Environment(
+        loader=jinja2.FileSystemLoader(path)
+    ).get_template(filename).render(context)
+
+
```
Comment (sangeethah) [inline tests/v2_validation/cattlevalidationtest/core/test_stress.py]: We should be able to reuse the validate_stack() from test_upgrade.py

**Comment by sangeethah** [comment]
@galal-hussein @moelsayed , Can we get the following additional checks done as well ?

1. Make the pre upgrade checks and post upgrade checks to include all the scenarios that have been dealt with in test_upgrade.py

2. Can we include few more helm commands for helm validation ?

File: tests/v2_validation/cattlevalidationtest/core/test_stress.py (hunk 3 above)
Comment (galal-hussein) [inline tests/v2_validation/cattlevalidationtest/core/test_stress.py]: Done.

Comment (galal-hussein) [inline tests/v2_validation/cattlevalidationtest/core/test_stress.py]: Done.

File: tests/v2_validation/cattlevalidationtest/core/common_fixtures.py (hunk 1 above)
Comment (galal-hussein) [inline tests/v2_validation/cattlevalidationtest/core/common_fixtures.py]: I figured out the problem, so basically random library in python uses time in seconds on the running machine to seed the generator, that's why there were failures in parallel jobs, we might put another variable in the random.seed(), in the mean time we added a work around in Jenkins job to sleep random number less < 99 seconds  before starting the job.

**CC** @moelsayed

File: tests/v2_validation/cattlevalidationtest/core/test_stress.py (hunk 2 above)
Comment (galal-hussein) [inline tests/v2_validation/cattlevalidationtest/core/test_stress.py]: Done, and i added modify stack as well

**Pull request closed without merging**